packages/analytics/amplify_analytics_pinpoint/lib/src/legacy_native_data_provider/data_provider.ios.dart

```diff
@@ -0,0 +1,36 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:async';
+
+import 'package:amplify_analytics_pinpoint/src/legacy_native_data_provider/pigeon_data_provider.ios.g.dart';
+import 'package:amplify_analytics_pinpoint_dart/amplify_analytics_pinpoint_dart.dart';
+
+/// {@macro amplify_analytics_pinpoint.flutter_legacy_native_data_provider}
+class DataProviderIos implements LegacyNativeDataProvider {
```

Comment:
They use the same class name but the implementations are different.  See: 


import 'package:amplify_analytics_pinpoint/src/legacy_native_data_provider/pigeon_data_provider.android.g.dart';

import 'package:amplify_analytics_pinpoint/src/legacy_native_data_provider/pigeon_data_provider.ios.g.dart';

Reply:
`pigeon_data_provider.android.g.dart` and `pigeon_data_provider.ios.g.dart` seem to have the exact same code though. You can combine the following:
- `pigeon_config_ios.dart` and `pigeon_config_android.dart`
- `data_provider.ios.dart` and `data_provider.android.dart`
- All `flutter_legacy_native_data_provider*` files

Typically when pigeon is being used, we want to expose the same APIs from each native platform (`initialize` and `getEndpointId` in this case). The dart code can be platform agnostic, and the swift/java code can be specific to the platform.

We have some rare cases where we expose a completely different set of APIs on each platform. You may have been following one of these as an example. For example, in secure_storage we have the need to interact with encrypted storage on android, whereas on iOS we only need to set/read a bool value from unecrypted storage. Since these are totally different use cases that require a different set of APIs, there are .ios and .android files for pigeon. This is not one of those rare examples where the dart code also needs to be different on each platform.

Reply:
Thanks for clarifying that Jordan.  Yes I followed the secure storage example and should have combined them. 

However now that secure storage is working I have removed the ios pigeon part.